### PR TITLE
Improve documentation on Editor Support

### LINKED ISSUE #1894: Documentation on Essential Editor Features for OCaml Beginners
# Update Documentation on using editors with OCaml

The documentation on setting up editors for OCaml, [Configure your editor](https://ocaml.org/docs/configuring-your-editor), is quite summarized and does not cover some useful features of the editor (VS Code), especially for beginners, whom I assume the article is directed towards.
This proposal is to enhance the documentation on ocaml.org to include detailed information. While these are not specific to OCaml, it will be very helpful for those beginners who are introduced to programming with OCaml as their first language.

The main things to include are listed below:
### Hovering for Type Information:
Description: The ability to hover over variables or functions to view their type information is a fundamental feature that aids in understanding code without having to navigate elsewhere.
Example: Hovering over a variable should display its type information in a tooltip.

### Ctrl + Hovering for Go-to-Definition:
Description: This feature allows users to navigate directly to the definition of a variable or function by holding down the Ctrl key and hovering over the identifier.
Example: Ctrl + Hovering over a function should provide a clickable link or open the file where the function is defined.

### Ctrl + P + [Command]:
Description: The ability to use a shortcut (Ctrl + P on Linux; find corresponding keys for Windows and Mac) followed by a specific command (e.g., OCaml) to find and run commands within the development environment.
Example: Ctrl + P + OCaml should allow users to quickly find and execute OCaml-specific commands.

### Click on Lightbulb for Code Actions:
Description: The lightbulb icon, often displayed next to code snippets, indicates available code actions. Clicking on it should reveal a list of suggestions for improving or modifying the code.
Example: Clicking on the lightbulb should show options like fixing syntax errors, optimizing imports, etc.

cc @pitag-ha

### PR DESCRIPTION
**Pull request by PizieDust**
closes #1894 

This PR contains some documentation changes for the editor support pages on [ocaml.org](https://ocaml.org)
- https://ocaml.org/docs/set-up-editor
- https://ocaml.org/docs/configuring-your-editor

The following changes were made:
- Replace images in favour of gifs. Gifs are more informative and can demonstrate much more than a static image. 
- Examples cover all editors (VSCode, Vim and Emacs)
- Links included for further information, such as documentation of Merlin features for vim and emacs. 
- Use of relative urls instead of absolute urls (I believe should be the standard for static assets)

cc @voodoos @pitag-ha

### CODE REVIEW

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 1]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
...
 
-```
+```shell
 $ opam install ocaml-lsp-server ocamlformat
 ```
 
-Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., 5.1.0 from the list. Additional information is available by hovering over symbols in your program:
+Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., `5.1.0` from the list. 
+
+### Editor features at your disposal
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
### Editor Features at Your Disposal
```

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 2]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
 
 # Configuring Your Editor
 
-While the toplevel is great for interactively trying out the language, we will shortly need to write OCaml files in an editor. We already installed the tools required to enhance Merlin, our editor of choice with OCaml support. Merlin provides all features such as "jump to definition," "show type," and `ocaml-lsp-server`, a server delivers those features to the editor through the LSP server.
+While the toplevel is great for interactively trying out the language, we will shortly need to write OCaml files in an editor. We already installed the tools required to enhance Merlin, our editor of choice with OCaml support. Merlin provides all features such as "jump to definition," "show type," and `ocaml-lsp-server`, a server which delivers those features to the editor through the LSP server.
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
While the toplevel is great for interactively trying out the language, we will shortly need to write OCaml files in an editor. We already installed the tools required to enhance Merlin, our editor of choice with OCaml support. Merlin provides all features such as "jump to definition," "show type," and `ocaml-lsp-server`, a server that delivers those features to the editor through the LSP server.
```

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 3]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
...
+After installing Merlin above, instructions will be printed on how to link Merlin with your editor. If you do not have them visible, just run this command:
 
-```
+```shell
 $ opam user-setup install
 ```
+
+### Talking to Merlin
+
+#### Getting Type information
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
#### Getting Type Information
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 4]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
 The extension depends on OCaml LSP and ocamlformat. To install them in your switch, you can run:
 
 ```shell
 $ opam install ocaml-lsp-server ocamlformat
 ```
 
-Upon first loading an OCaml source file, you may be prompted to select the
-toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0
-from the list. Additional information is available by hovering over symbols in your program:
+Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0 from the list.
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0, from the list. 
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 5]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
 $ opam install ocaml-lsp-server ocamlformat
 ```
 
-Upon first loading an OCaml source file, you may be prompted to select the
-toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0
-from the list. Additional information is available by hovering over symbols in your program:
+Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0 from the list. 
 
-![Visual Studio Code](/media/tutorials/vscode.png)
+### Editor features at your disposal
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
### Editor Features at Your Disposal
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 6]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
 
 ```shell
 $ opam user-setup install
-```
\ No newline at end of file
+```
+
+### Talking to Merlin
+
+#### Getting Type information
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
#### Getting Type Information
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 7]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
+
+![VSCode Ctrl click](/media/tutorials/vscode-ctrl-click.gif)
+
+If you hold down the <kbd>Ctrl</kbd> key while hovering, the code appears as a clickable link which if clicked takes you to the file where the implementation is. This can be great if you want to understand how a piece of code works under the hood. In this example, hovering and `Ctrl + Clicking` over the `peek` method of the `Queue` module will take you to the definiton of the `peek` method itself and how it is implemented.
+
+#### 3) OCaml commands with `Ctrl + Shift + p`:
+
+![VSCode OCaml Commands](/media/tutorials/vscode-ocaml-commands.gif)
+
+Pressing the key combination <kbd>Ctrl + Shift + p</kbd> opens a modal dialog at the top. If you type the word `ocaml`, you will be presented with a list of various OCaml commands at your disposal which can be used for different purposes.
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
Pressing the key combination <kbd>Ctrl + Shift + P</kbd> opens a modal dialog at the top. If you type the word `ocaml`, you will be presented with a list of various OCaml commands at your disposal which can be used for different purposes.
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 8]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
+
+This is a great feature that let's you see type information of any OCaml variable or function. All you have to do is place your cursor over the code and it will be displayed in the tooltip.
+
+#### 2) Jump to definitions with `Ctrl + Click`:
+
+![VSCode Ctrl click](/media/tutorials/vscode-ctrl-click.gif)
+
+If you hold down the <kbd>Ctrl</kbd> key while hovering, the code appears as a clickable link which if clicked takes you to the file where the implementation is. This can be great if you want to understand how a piece of code works under the hood. In this example, hovering and `Ctrl + Clicking` over the `peek` method of the `Queue` module will take you to the definiton of the `peek` method itself and how it is implemented.
+
+#### 3) OCaml commands with `Ctrl + Shift + p`:
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
#### 3) OCaml Commands With `Ctrl + Shift + P`:
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 9]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
-![Visual Studio Code](/media/tutorials/vscode.png)
+### Editor features at your disposal
+If your editor is setup correctly, here are some important features you can begin using to your advantage:
+#### 1) Hovering for type information: 
+
+![VSCode Hovering](/media/tutorials/vscode-hover.gif)
+
+This is a great feature that let's you see type information of any OCaml variable or function. All you have to do is place your cursor over the code and it will be displayed in the tooltip.
+
+#### 2) Jump to definitions with `Ctrl + Click`:
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
#### 2) Jump to Definitions With `Ctrl + Click`:
```

File: data/tutorials/platform/bp_06_configuring_your_editor.md [hunk 10]
```diff
@@ -14,39 +14,77 @@ OCaml has plugins for many editors, but the most actively maintained are for Vis
...
 
-Upon first loading an OCaml source file, you may be prompted to select the
-toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0
-from the list. Additional information is available by hovering over symbols in your program:
+Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., 4.14.0 from the list. 
 
-![Visual Studio Code](/media/tutorials/vscode.png)
+### Editor features at your disposal
+If your editor is setup correctly, here are some important features you can begin using to your advantage:
+#### 1) Hovering for type information:
```
Comment (christinerose) [inline data/tutorials/platform/bp_06_configuring_your_editor.md]:
```suggestion
#### 1) Hovering for Type Information: 
```

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 11]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
...
+![VSCode Ctrl click](/media/tutorials/vscode-ctrl-click.gif)
 
-For Windows
+If you hold down the <kbd>Ctrl</kbd> key while hovering, the code appears as a clickable link which if clicked takes you to the file where the implementation is. This can be great if you want to understand how a piece of code works under the hood. In this example, hovering and `Ctrl + Clicking` over the `peek` method of the `Queue` module will take you to the definiton of the `peek` method itself and how it is implemented.
 
-If you used the **Diskuv OCaml (DKML) installer** you will need to:
-    1. Go to **File > Preferences > Settings** view (or press **Ctrl**)
-    2. Select **User > Extensions > OCaml Platform**
-    3. **Uncheck OCaml: Use OCaml Env.** That's it!
+#### 3) OCaml commands with `Ctrl + Shift + p`:
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
#### 3) OCaml Commands With `Ctrl + Shift + P`:
```

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 12]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
...
+
+![VSCode OCaml Commands](/media/tutorials/vscode-ocaml-commands.gif)
+
+Pressing the key combination <kbd>Ctrl + Shift + p</kbd> opens a modal dialog at the top. If you type the word `ocaml`, you will be presented with a list of various OCaml commands at your disposal which can be used for different purposes.
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
Pressing the key combination <kbd>Ctrl + Shift + P</kbd> opens a modal dialog at the top. If you type the word `ocaml`, you will be presented with a list of various OCaml commands at your disposal which can be used for different purposes.
```

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 13]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
...
+
+### Editor Features at Your Disposal
+If your editor is setup correctly, here are some important features you can begin using to your advantage:
+#### 1) Hovering for type information: 
+
+![VSCode Hovering](/media/tutorials/vscode-hover.gif)
+
+This is a great feature that let's you see type information of any OCaml variable or function. All you have to do is place your cursor over the code and it will be displayed in the tooltip.
+
+#### 2) Jump to definitions with `Ctrl + Click`:
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
#### 2) Jump to Definitions With `Ctrl + Click`:
```

File: data/tutorials/getting-started/2_00_editor_setup.md [hunk 14]
```diff
@@ -8,42 +8,82 @@ category: "Tooling"
...
+```shell
 $ opam install ocaml-lsp-server ocamlformat
 ```
 
-Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., 5.1.0 from the list. Additional information is available by hovering over symbols in your program:
+Upon first loading an OCaml source file, you may be prompted to select the toolchain in use. Pick the version of OCaml you are using, e.g., `5.1.0` from the list. 
+
+### Editor Features at Your Disposal
+If your editor is setup correctly, here are some important features you can begin using to your advantage:
+#### 1) Hovering for type information:
```
Comment (christinerose) [inline data/tutorials/getting-started/2_00_editor_setup.md]:
```suggestion
#### 1) Hovering for Type Information: 
```

**Review by christinerose — APPROVED** [review]

**Review by sabine — COMMENTED** [review]
Hey @PizieDust, this is great!

However, one thing I'll have to ask is that we do not use company names in these tutorials.

Otherwise, in the sake of fairness for all participants of the OCaml ecosystem, we will have to allow this across the board in content contributed for the OCaml.org documentation. And I do not think we want that.

**Comment by PizieDust** [comment]
> Hey @PizieDust, this is great!
> 
> However, one thing I'll have to ask is that we do not use company names in these tutorials.
> 
> Otherwise, in the sake of fairness for all participants of the OCaml ecosystem, we will have to allow this across the board in content contributed for the OCaml.org documentation. And I do not think we want that.

Oh, thanks for pointing that out. I'll replace the string with something generic.

**Review by sabine — APPROVED** [review]
LGTM! Thanks @PizieDust! :sparkles:

**Pull request merged**